### DISCUSSION TITLE
evaluatie output

### DISCUSSION THREAD

**Discussion by PythonSPC** [dodona-edu/feedback-and-support#54] [Questions - Vragen] [answered]
Ik tracht nieuwe oefeningen aan te maken en had 2 vragen in verband met de evaluatie voor de output:
1. Als uitvoer moet er een zin afgedrukt worden waarin de waarde van een berekende variabele staat, dit is een variabele van het type float. In mijn evaluatie voorzag ik vb. 4.5 maar bij het testen ervan wordt dit vb. 4.50005 en is dit niet correct. Moet ik mijn oplossingen testen in dodona en deze waardes overnemen of is er een andere manier om dit op te vangen?
2. Aangezien leerlingen al wel eens fouten maken in de zin zelf, is er een mogelijkheid om vb. een jokerteken te gebruiken. vb. het antwoord moet zijn Dat is dan €5 . (--> 5 werd uitgerekend en moet in de zin voorkomen maar de zin Je moet €5 betalen zou ook juist gerekend moeten worden).

Alvast bedankt.

**Comment by jorg-vr** [comment, marked as answer]
1. Het is niet eenvoudig mogelijk om binnen een string om te gaan met floats. In dit soort opgave wordt vaak gevraagd aan de studenten om af te ronden op een gegeven aantal cijfers na de comma. Zo weet je als schrijver van de test ook welke waarde je moet verwachten.

2. We ondersteunen momenteel geen jokers in strings. Als de zin zelf niet belangerijk is zou ik hen vragen om rechtstreeks de variabele terug te geven, zodat je enkel hierop test.

Beide gevallen kan je wel oplossen via een [orakelfunctie](https://docs.dodona.be/nl/guides/exercises/testsuites/#eigen-orakelfunctie-custom-check). Dit is een gevorderde feature van TESTed die jou toelaat om eigen code te schrijven om het resultaat te controleren.

**Reply by PythonSPC** [reply]
Bedankt voor de snelle respons.
Ik volg om te starten je tips op en bestudeer later de orakelfunctie nog.
mvg
